### ISSUE TITLE
I receive "broken pipe" error when trying to "set" in async version. Crate version 0.21.5

### ISSUE THREAD

**Issue by formiat** [redis-rs/redis-rs#628]
I tried `crate::aio::Connection` (`fn get_async_connection()`) and `crate::aio::MultiplexedConnection` (`fn get_multiplexed_async_connection()`). Getting "**broken pipe**" error in all cases. Non-async version works good.

Maybe it was broken in some version? Is there version with working async connection?

**Comment by djc** [comment]
We don't have any other bug reports like this and I'm pretty sure 0.21.5 is in wide use (I use it in my projects with an async connection, too). Therefore, I'm pretty sure it has something to do with your environment. Perhaps check the connection string is getting passed in correctly?

**Comment by l-monninger** [comment]
In a service of mine, I'm getting somewhat too frequent broken pipes when using `zrange` and a `MultiplexedConnection`. These only occur on highly concurrent (~2/ms) requests on the same key and high traffic (~100/ms) all around.

I don't think this is a bug, and it may not be the same as your issue. But, I do think Rust makes it pretty easy to wind up with throughput to Redis that is simply too high. 

That being said, it would be nice to have some kind of guide for figuring out whether there's room to improve the Rust or whether we're running into the fundamental limits of Redis.

**Comment by djc** [comment]
I'm not sure there's much I can do for you. It seems unlikely to me that redis-rs would be causing such issues, I don't think we initialize any "broken pipe" errors. As such, I would guess you're running into fundamental limits of Redis, but I don't know Redis well enough at that level to be able to say more. I would probably dig more into logging and tracing to see if you can correlate why/when/where you see "broken pipe" errors occurring (and I'd be happy to be informed if you figure out where the bottleneck is).

What do you mean by "Rust makes it pretty easy to wind up with throughput  to Redis that is simply too high."?

**Comment by djc** [comment]
@gkorland any suggestions here?

**Comment by oranagra** [comment]
stepping in to mention i have no clue, but i'm guessing it comes from the system.
I think that in order to know more, you must get more info.
maybe set redis to debug log level to see all the connections being closed, or maybe use `strace` on the client to see what exactly it does and which error it gets.

**Comment by l-monninger** [comment]
> What do you mean by "Rust makes it pretty easy to wind up with throughput to Redis that is simply too high."?

Rust is fast.

**Comment by getong** [comment]
Very old redis instance, 5.0.6
rust 1.80
redis-rs 0.27 
use with async code
warning msg:
```
Failed to retrieve members from Redis set key_name: broken pipe
```

It might be the problem with old version redis. Does redis-rs still support redis 5 and 6?

**Comment by nihohit** [comment]
Redis-rs should work with older versions of redis. Have you looked into the suggestions of his to get the source of the error proposed in the discussion on this issue?

**Comment by getong** [comment]
> Redis-rs should work with older versions of redis. Have you looked into the suggestions of his to get the source of the error proposed in the discussion on this issue?
Yes, it might be the haproxy config error. The redis is used with haproxy

**Comment by getong** [comment]
> Redis-rs should work with older versions of redis. Have you looked into the suggestions of his to get the source of the error proposed in the discussion on this issue?

Yes, it might be the haproxy config error. The redis is used with haproxy
